### DISCUSSION TITLE
Use OpenFontRender to render button labels

### DISCUSSION THREAD

**Discussion by marcelstoer** [Bodmer/TFT_eSPI#2566] [Q&A - Fonts] [answered]
Has anyone used https://github.com/Bodmer/OpenFontRender to render button labels?

The following screen shot shows why this might be desirable.

![DSC07478](https://user-images.githubusercontent.com/624195/230614820-5644c6c7-0e16-4311-843f-9eabf4970ef1.JPG)

A few words rendered by an OpenFontRender example (example TTF) next to a button from this library drawn using `button.initButton(&tft, 400, 30, 141, 43, 0xFFFF, 0x2A54, 0xFFFF, "Setup", 2);` and a simple text `tft.drawString("Hello World!", 10, 0, 4);`. Wouldn't it be nice if the button label rendered in high-def? The obvious workaround might be to give the button a `""` label and then draw over it?

**Comment by Bodmer** [comment, marked as answer]
Smooth font text using vlw files is supported, so that is an option, but as you found TTF fonts are not.

Yes, you could overprint the button but this may not be so simple if you have an inverting button.

TTF font support is quite new and new capabilities may be added at some point but I have other priorities at the moment.

**Reply by marcelstoer** [reply]
Thanks for the VLW hint! I'll check the `/examples/Smooth Fonts/*` examples and the related VLW discussions at https://github.com/Bodmer/TFT_eSPI/discussions?discussions_q=vlw+

**Reply by marcelstoer** [reply]
Sorry for coming back with this. Is this a misunderstanding or do I not see the obvious? I went through a couple of VLW examples but I still don't see how VLW fonts could be used for button labels?

**Reply by Bodmer** [reply]
The buttons will use the currently loaded or selected font. So for example:

... add a vlw font in FLASH to your sketch - see examples
Then in sketch:
tft.loadFont(font_name);
... init button
...draw button

**Reply by marcelstoer** [reply]
Oh, I see. The `uint8_t textsize` parameter in `initButton()` is then simply ignored, understood.
